sync2/handler2/handler.go

```diff
@@ -370,20 +370,24 @@ func (h *Handler) Accumulate(ctx context.Context, userID, deviceID, roomID strin
 	return nil
 }
 
-func (h *Handler) Initialise(ctx context.Context, roomID string, state []json.RawMessage) ([]json.RawMessage, error) {
+func (h *Handler) Initialise(ctx context.Context, roomID string, state []json.RawMessage) error {
 	res, err := h.Store.Initialise(roomID, state)
 	if err != nil {
 		logger.Err(err).Int("state", len(state)).Str("room", roomID).Msg("V2: failed to initialise room")
 		internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(err)
-		return nil, err
+		return err
 	}
-	if res.AddedEvents {
+	if res.ReplacedExistingSnapshot {
+		h.v2Pub.Notify(pubsub.ChanV2, &pubsub.V2InvalidateRoom{
+			RoomID: roomID,
+		})
+	} else if res.AddedEvents {
```

Comment:
Are they mutally exclusive? If so, an enum would be better than 2 bools.

Reply:
They are not mutually exclusive (though I think the question you mean to ask is "are they orthogonal?").

Reply:
What I really want is a sum type to be able to express three distinct states:

```
Noop
NewSnapshot(int64 snapID)
ReplacedSnapshot(int64 newSnapID)
```

I'm tempted to leave this as it is tbh.